### PR TITLE
Create fake data for users and interviews

### PR DESCRIPTION
**Pull request by abedshamia**

### CODE REVIEW

**Review by Mahmoud-Ahmad2 — APPROVED** [review]

File: server/database/fakeData/Users.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+    "user": [
+        {
+            "id": 1,
+            "full_name": "Jane Doe",
+            "email": "jane@gmail.com",
+            "password": "$2a$10$KXdWNpCUVb9U8ePldP419undqNuiA321ZfM4W4vDbs9CpnSBPpCDC",
+            "bio": "I'm a web developer",
+            "cv": "link to cv",
+            "profile_pic": "https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50",
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Users.json:10]: Its `name` not `full_name`, and `profile_picture` not `profile_pic`

File: server/database/fakeData/Schedule.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+    "schedule": [
+        {
+            "language": ["JavaScript", "PHP"],
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Schedule.json:4]: Its `languages` not `language`, and it is just a string not an array

File: server/database/fakeData/Schedule.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            "specialization": ["Frontend", "Backend"],
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Schedule.json:5]: specialization is string not an array

File: server/database/fakeData/Interviewers.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
+{
+    "interviewer":[
+        {
+            "userId": "2",
+            "languages": [
+                "JavaScript",
+                "PHP"
+            ],
+            "specialization": "Frontend",
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Interviewers.json:9]: Check the options in the schema

File: server/database/fakeData/Interviewers.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            "interviews": [
+                {
+                    "id": "1",
+                    "intervieweeId": 1,
+                    "date": "2022-04-28",
+                    "time": 12,
+                    "language": "JavaScript",
+                    "specialization": "Frontend",
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Interviewers.json:17]: Check their options from the schema

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Comment by abedshamia** [comment]
I couldnt reslove the issues of the enum with the build. Need assistance with it.

File: server/database/build.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      await Interviewee.create(interviewee);
+    });
+
+    interviewersData.interviewer.forEach(async (interviewer) => {
+      await Interviewer.create(interviewer);
+    });
+
+    console.log('Fake data created successfully');
+  } catch (err) {
+    console.log(err);
```
Comment (Raghad-Mezied) [inline server/database/build.ts:41]: Threw an error here and delete the console.

File: server/database/build.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  }
+};
+
+startDb().then(async () => {
+  await createFakeData();
+});
```
Comment (Raghad-Mezied) [inline server/database/build.ts:47]: We don't want to invoke this function in the test case, so use `startDb` inside your function and add a condition to invoke `createFakeData`.

File: server/database/fakeData/Schedule.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
+{
+    "schedule": [
+        {
+            "languages": "JS, PHP",
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Schedule.json:4]: This should be a string which means just one value(lang).

File: server/database/fakeData/Schedule.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+            "specialization": "Frontend, Backend",
```
Comment (Raghad-Mezied) [inline server/database/fakeData/Schedule.json:5]: The same as languages

File: server/database/models/interviewee.ts [hunk 10]
```diff
@@ -23,17 +23,18 @@ const interviewee = new Schema({
           required: true,
         },
         language: {
-          type: String,
+          type: [String],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewee.ts:26]: This should be just one lang so its String

File: server/database/models/interviewee.ts [hunk 11]
```diff
@@ -23,17 +23,18 @@ const interviewee = new Schema({
...
           enum: ['JS', 'PHP', 'C++', 'C#', 'RUBY', 'PYTHON', 'JAVA', 'C', 'GO'],
           required: true,
         },
         specialization: {
-          type: String,
+          type: [String],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewee.ts:31]: Type String

File: server/database/models/interviewee.ts [hunk 12]
```diff
@@ -23,17 +23,18 @@ const interviewee = new Schema({
...
           enum: ['FRONTEND', 'BACKEND', 'DEVOPS', 'SECURITY', 'DATA STRUCTURE', 'FULL STACK'],
           required: true,
         },
         questionCategory: {
-          type: Array,
+          type: [String],
+          enum: ['Technical', 'Behavioural'],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewee.ts:37]: Add these to enum values `Technical, Analytical, Algorithms, System Design`

File: server/database/models/interviewer.ts [hunk 13]
```diff
@@ -7,12 +7,12 @@ const interviewer = new Schema({
...
   },
   languages: {
-    type: Array,
+    type: [String],
     enum: ['JS', 'PHP', 'C++', 'C#', 'RUBY', 'PYTHON', 'JAVA', 'C', 'GO'],
     required: true,
   },
   specialization: {
-    type: String,
+    type: [String],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewer.ts:15]: Type string

File: server/database/models/interviewer.ts [hunk 14]
```diff
@@ -33,18 +33,20 @@ const interviewer = new Schema({
           required: true,
         },
         language: {
-          type: String,
+          type: [String],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewer.ts:36]: type String

File: server/database/models/interviewer.ts [hunk 15]
```diff
@@ -33,18 +33,20 @@ const interviewer = new Schema({
...
           enum: ['JS', 'PHP', 'C++', 'C#', 'RUBY', 'PYTHON', 'JAVA', 'C', 'GO'],
           required: true,
         },
         specialization: {
-          type: String,
+          type: [String],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewer.ts:41]: type String

File: server/database/models/interviewer.ts [hunk 16]
```diff
@@ -33,18 +33,20 @@ const interviewer = new Schema({
...
           enum: ['FRONTEND', 'BACKEND', 'DEVOPS', 'SECURITY', 'DATA STRUCTURE', 'FULL STACK'],
           required: true,
         },
         questionCategory: {
-          type: Array,
+          type: [String],
           required: true,
+          enum: ['Technical', 'Behavioural'],
```
Comment (Raghad-Mezied) [inline server/database/models/interviewer.ts:48]: Edit options here too

File: server/database/models/user.ts [hunk 17]
```diff
@@ -28,7 +28,7 @@ const user = new Schema({
     required: false,
   },
   level: {
-    type: String,
+    type: [String],
```
Comment (Raghad-Mezied) [inline server/database/models/user.ts:31]: type String

File: server/database/models/schedule.ts [hunk 18]
```diff
@@ -9,10 +9,14 @@ const schedule = new Schema({
     type: String,
     required: false,
   },
+  questionCategory: {
+    type: String,
+    required: false,
+  },
```
Comment (Raghad-Mezied) [inline server/database/models/schedule.ts:15]: Add options for `languages`, `specialization`, and `questionCategory` in this model.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by manar-abed — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Pull request merged**